vendor/font-awesome-as-a-crate/build.rs

```diff
@@ -0,0 +1,54 @@
+use std::{
+    env,
+    fs::{read_dir, File},
+    io::{Read, Write},
+    path::Path,
+};
+
+fn main() {
+    println!("cargo:rustc-cfg=font_awesome_out_dir");
+    write_fontawesome_sprite();
+}
+
+fn write_fontawesome_sprite() {
+    let dest_path = Path::new(&env::var("OUT_DIR").unwrap()).join("fontawesome.rs");
+    let mut dest_file = File::create(&dest_path).unwrap();
+    dest_file
+        .write_all(b"fn fontawesome_svg(dir:&str,file:&str)->&'static str{match(dir,file){")
+        .expect("fontawesome fn write");
+    for dirname in &["brands", "regular", "solid"] {
+        let dir = read_dir(Path::new("fontawesome-free-5.14.0-web/svgs").join(dirname)).unwrap();
+        let mut data = String::new();
+        for file in dir {
+            let file = file.expect("fontawesome directory access");
+            let filename = file
+                .file_name()
+                .into_string()
+                .expect("fontawesome filenames are unicode");
+            let mut file = File::open(file.path()).expect("fontawesome file access");
+            data.clear();
+            file.read_to_string(&mut data)
+                .expect("fontawesome file read");
+            // if this assert goes off, add more hashes here and in the format! below
+            assert!(
+                data.find("###").is_none(),
+                "file {} breaks raw string",
+                filename,
+            );
+            dest_file
+                .write_all(
+                    format!(
+                        r####"("{dirname}","{filename}")=>r###"{data}"###,"####,
+                        data = data,
+                        dirname = dirname,
+                        filename = filename.replace(".svg", ""),
+                    )
+                    .as_bytes(),
+                )
+                .expect("write fontawesome file");
```

Comment:
Wow, this is _clever_ 😆 I don't know whether to be impressed or horrified, I'll go with impressed.